packages/cdk-assets/lib/private/handlers/container-images.ts

```diff
@@ -45,6 +50,12 @@ export class ContainerImageAssetHandler implements IAssetHandler {
     this.host.emitMessage(EventType.UPLOAD, `Push ${imageUri}`);
     if (this.host.aborted) { return; }
     await this.docker.tag(localTagName, imageUri);
+
+    if (cdkDockerCredentialsConfigured) {
+      this.docker.createCleanConfig();
```

Comment:
Maybe a name like `resetAuthPlugins()` or something, to make it very clear why we're calling that here?

Reply:
Sure.

This actually brings up a design decision I made I'm not 100% sure about.

`resetAuthPlugins` could do what it does now, which is to create and use an empty config directory. Instead, we could have it simply *not* use the cdk-config directory and default to whatever's on the system. The latter is slightly more backwards-compatible, as it mimics the current behavior; the former felt like a stronger guarantee of behavior, but maybe for the wrong reasons. Thoughts?

Reply:
Let's fall back to default config. For the current design target it's the same.

Do we need to clean a cache of login credentials though? What if Docker already used a credential helper to fetch an initial set? Will that work properly?

(Hmmm I guess the `docker login` will override the previous creds?)

Reply:
> Do we need to clean a cache of login credentials though? What if Docker already used a credential helper to fetch an initial set? Will that work properly? (Hmmm I guess the docker login will override the previous creds?)

Docker stores the fact that it has auth'd in the config directory, so if you change directories, the auth cache is busted and you reset. Verified in the live pipeline -- `docker logout` not necessary.